src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -50,30 +51,87 @@
- * &#64;SuppressWarnings("foo") enum E { }
- * &#64;SuppressWarnings("foo") InputSuppressWarningsFilter() { }
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;MemberName&quot;/&gt;
+ *   &lt;module name=&quot;ConstantName&quot;/&gt;
+ *   &lt;module name=&quot;ParameterNumber&quot;/&gt;
+ *   &lt;module name=&quot;NoWhitespaceAfter&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private int K; // violation
+ *     &#64;SuppressWarnings({"membername"})
+ *     private int J; // ok
+ *
+ *     private static final int i = 0; // violation
+ *     &#64;SuppressWarnings("checkstyle:constantname")
+ *     private static final int m = 0; // ok
+ *
+ *     public void needsLotsOfParameters (int a, // violation
+ *       int b, int c, int d, int e, int f, int g, int h) {
+ *       // ...
+ *     }
+ *
+ *     &#64;SuppressWarnings("paramnum")
+ *     public void needsLotsOfParameters1 (int a, // ok
+ *       int b, int c, int d, int e, int f, int g, int h) {
+ *       //  ...
+ *     }
+ *
+ *    private int [] ARR; // 2 violations
+ *    &#64;SuppressWarnings("all")
+ *    private int [] ARRAY; // ok
+ * }
  * </pre>
  * <p>
- * Some real check examples:
- * </p>
- * <p>
- * This will prevent from invocation of the MemberNameCheck:
+ * This will prevent selected violations from MemberNameCheck:
  * </p>
  * <pre>
- * &#64;SuppressWarnings({"membername"})
- * private int J;
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;MemberName&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private int K; // violation
+ *     &#64;SuppressWarnings({"membername"})
+ *     private int J; // violation suppressed
+ *
+ * }
  * </pre>
  * <p>
  * You can also use a {@code checkstyle} prefix to prevent compiler from
- * processing these annotations. For example this will prevent ConstantNameCheck:
+ * processing these annotations. For example this will prevent
+ * selected violations from ConstantNameCheck:
  * </p>
  * <pre>
- * &#64;SuppressWarnings("checkstyle:constantname")
- * private static final int m = 0;
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;ConstantName&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private static final int i = 0; // violation
+ *     &#64;SuppressWarnings("checkstyle:constantname")
+ *     private static final int m = 0; // violation suppressed
+ *
+ * }
  * </pre>
  * <p>
  * The general rule is that the argument of the {@code @SuppressWarnings} will be
```

Comment:
> matched against class name of the checker in lower case and without {@code Check}
> suffix if present.

Please add an example of this, matching against Name and NameCheck.

Reply:
@Kevin222004 , please use both names (with and wihtout "Check" suffix) 
`SuppressWarnings("checkstyle:constantname")` and `SuppressWarnings("checkstyle:constantnamecheck")` 
to show how it works.

if something does not work - do not extend and lets create separarte issue on this.

Reply:
@romani  ```SuppressWarnings("checkstyle:constantname")``` without "Check" suffix it is working 
```
$ cat Test.java
class Test {

  private static final int i = 0;

  @SuppressWarnings("checkstyle:constantname")
  private static final int m = 0;
}


$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	
   
<module name="TreeWalker">
   <module name="ConstantName">
   </module>
   
   <module name="SuppressWarningsHolder">
	</module>
</module>

   <module name="SuppressWarningsFilter" />
</module>

$ java -jar /home/kevin/Downloads/checkstyle-10.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/C_Style/Test.java:3:28: Name 'i' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
Audit done.
Checkstyle ends with 1 errors.
```

but with "check" suffix it is not working 
```
$ cat Test.java
class Test {

  private static final int i = 0;

  @SuppressWarnings("checkstyle:constantnamecheck")
  private static final int m = 0;
}


$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	
   
<module name="TreeWalker">
   <module name="ConstantName">
   </module>
   
   <module name="SuppressWarningsHolder">
	</module>
</module>

   <module name="SuppressWarningsFilter" />
</module>

$ java -jar /home/kevin/Downloads/checkstyle-10.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/C_Style/Test.java:3:28: Name 'i' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
[ERROR] /home/kevin/Desktop/C_Style/Test.java:6:28: Name 'm' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
Audit done.
Checkstyle ends with 2 errors.
```

Reply:
Thanks a lot.

Please create separate issue and on this.

No need to update doc now.

Reply:
Please create ticket

Reply:
https://github.com/checkstyle/checkstyle/issues/11885

@Kevin222004 , Please do not resolve my review items, just reply done, links, and extra comment

Reply:
> Please create ticket

Ticket ? ,  What is it

Reply:
@Kevin222004 He is talking about new issue so we can track an issue and eventually resolve it in another PR.

Reply:
@rnveach, thanks 
it is been done at https://github.com/checkstyle/checkstyle/issues/11885